java/src/main/java/com/xgen/interview/ShoppingCart.java

```diff
@@ -30,7 +30,7 @@ public void printReceipt() {
 
         for (int i = 0; i < Array.getLength(keys) ; i++) {
             Integer price = pricer.getPrice((String)keys[i]) * contents.get(keys[i]);
-            Float priceFloat = new Float(new Float(price) / 100);
+            Float priceFloat = Float.valueOf(price) / 100;
```

Comment:
this is intentionally ugly, hoping that candidates will clean it up :)

Reply:
Whops, reverted